core/src/main/java/org/apache/struts2/result/ServletDispatcherResult.java

```diff
@@ -173,9 +166,28 @@ public void doExecute(String finalLocation, ActionInvocation invocation) throws
                 request.setAttribute("struts.view_uri", finalLocation);
                 request.setAttribute("struts.request_uri", request.getRequestURI());
 
+                // These attributes are required when forwarding to another servlet, see specification:
```

Comment:
I think setting these is the responsibility of the servlet container e.g. tomcat, inside the implementation of the `forward` method. We should not have to do it here at all.